File: sprint8/By languages/Java/lesson4code2.java
public static List<Integer> findAll(String text, String pattern) {
    List<Integer> occurrences = new ArrayList<>();
    int start = 0; // Начнём поиск с начала строки.
    // Найдём первое вхождение, если оно есть.
    while (true) {
        int pos = find(text, pattern, start);
        if (pos == -1) {
            break;
        }
        occurrences.add(pos); // Сохраним вхождение в список.
        start = pos + 1;       // И продолжим поиск, начиная с позиции, 
                               // следующей за только что найденной.
    }
    return occurrences;
}

File: sprint8/By languages/C#/lesson4code2.cs
public static List<int> FindAll(string text, string pattern)
{
    List<int> occurrences = new List<int>();
    int start = 0; // Начнём поиск с начала строки.
    // Найдём первое вхождение, если оно есть.
    int pos;
    while ((pos = Find(text, pattern, start)) != -1)
    {
        occurrences.Add(pos); // Сохраним вхождение в список.
        start = pos + 1;      // И продолжим поиск, начиная с позиции, 
                              // следующей за только что найденной.
    }
    return occurrences;
}